packages/mdc-select/component.ts

```diff
@@ -50,23 +50,25 @@ export class MDCSelect extends MDCComponent<MDCSelectFoundation> implements MDCR
   // Public visibility for this property is required by MDCRippleCapableSurface.
   root_!: HTMLElement; // assigned in MDCComponent constructor
 
+  // Public visibility for backward compatibility.
+  ripple!: MDCRipple | null;
+
   private menu_!: MDCMenu | null; // assigned in enhancedSelectSetup_()
-  private menuOpened_ = false;
+  private isMenuOpen_!: boolean; // assigned in initialize()
 
   // Exactly one of these fields must be non-null.
   private nativeControl_!: HTMLSelectElement | null; // assigned in initialize()
   private selectedText_!: HTMLElement | null; // assigned in initialize()
 
   private targetElement_!: HTMLElement; // assigned in initialize()
 
-  private hiddenInput_!: HTMLInputElement | null;
-  private leadingIcon_?: MDCSelectIcon;
-  private helperText_!: MDCSelectHelperText | null;
-  private menuElement_!: Element | null;
-  private ripple!: MDCRipple | null;
-  private lineRipple_!: MDCLineRipple | null;
-  private label_!: MDCFloatingLabel | null;
-  private outline_!: MDCNotchedOutline | null;
+  private hiddenInput_!: HTMLInputElement | null; // assigned in enhancedSelectSetup_()
+  private menuElement_!: Element | null; // assigned in enhancedSelectSetup_()
+  private leadingIcon_?: MDCSelectIcon; // assigned in initialize()
```

Comment:
Unlike the other properties, `leadingIcon_` isn't always assigned a value, because it's too complex to use a ternary:

https://github.com/material-components/material-components-web/blob/720bef02fe5d55cffe827614de89f6eac8b0526b/packages/mdc-select/component.ts#L117-L123